### PR TITLE
Feature/roi create multiple

### PR DESCRIPTION
**Pull request by tdlinh2712**
This PR:

- [x] allows users to draw ROI on multiple slices in one go. It will save the drawing process on the current slice when the user changes slice

- [x]  Improve drawing cursor - If the user drags the cursor from an empty pixel, the cursor will be removing pixels, otherwise filling pixels. Users can also choose to only fill pixels within the defined range or fill all empty pixels.

- [x] Some UI fixes for draw ROI window

Please stress-test this functionality, under many scenarios and circumstances.

### CODE REVIEW

File: src/Model/ISO2ROI.py [hunk 1]
```diff
@@ -192,7 +192,7 @@ def generate_roi(self, contours, progress_callback):
                 # Create the ROI(s)
                 for array in single_array:
                     rtss = ROI.create_roi(dataset_rtss, item,
-                                          array, dataset, "DOSE_REGION")
+                                          [{'coords': array, 'ds': dataset}], "DOSE_REGION")
```
Comment (Tamookk) [inline src/Model/ISO2ROI.py:195]:
```suggestion
                                          [{'coords': array, 'ds': dataset}],
                                          "DOSE_REGION")
```

PEP 8 compliance

File: src/Model/ROI.py [hunk 2]
```diff
@@ -132,7 +132,7 @@ def add_to_roi(rtss, roi_name, roi_coordinates, data_set):
     return rtss
 
 
-def create_roi(rtss, roi_name, roi_coordinates, data_set,
+def create_roi(rtss, roi_name, roi_list,
                rt_roi_interpreted_type="ORGAN"):
     """
```
Comment (Tamookk) [inline src/Model/ROI.py:137]: Docstrings should be updated

File: src/Model/ROI.py [hunk 3]
```diff
@@ -153,121 +152,128 @@ def create_roi(rtss, roi_name, roi_coordinates, data_set,
     for key, value in existing_rois.items():
         if value["name"] == roi_name:
             roi_exists = True
+    for roi_info in roi_list:
+        data_set = roi_info['ds']
+        roi_coordinates = roi_info['coords']
+        if not roi_exists:
+            rtss = add_new_roi(rtss, roi_name, roi_coordinates, data_set, rt_roi_interpreted_type)
```
Comment (Tamookk) [inline src/Model/ROI.py:159]:
```suggestion
            rtss = add_new_roi(rtss, roi_name, roi_coordinates, data_set,
                               rt_roi_interpreted_type)
```

PEP 8 compliance

File: src/Model/ROI.py [hunk 4]
```diff
@@ -153,121 +152,128 @@ def create_roi(rtss, roi_name, roi_coordinates, data_set,
...
-                                     roi_number)
-            ROI_observations.add_new(Tag("RTROIInterpretedType"), 'CS',
-                                     rt_roi_interpreted_type)
-            ROI_observations.add_new(Tag("ROIInterpreter"), 'CS',
-                                     "")
-
-        original_roi_observation_sequence.extend(rt_roi_observations_sequence)
-        rtss.add_new(Tag("RTROIObservationsSequence"), "SQ",
-                     original_roi_observation_sequence)
+def add_new_roi(rtss, roi_name, roi_coordinates, data_set, rt_roi_interpreted_type):
```
Comment (Tamookk) [inline src/Model/ROI.py:168]:
```suggestion
def add_new_roi(rtss, roi_name, roi_coordinates, data_set,
                rt_roi_interpreted_type):
```

PEP 8 compliance. Also, need docstrings for this function.

File: src/View/mainpage/DrawROIWindow/DrawBoundingBox.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        self.drawing = False
+        self.start_x = None
+        self.start_y = None
+        self.end_x = None
+        self.end_y = None
+        self.box = None
+
+    def draw_bounding_box(self, new_box=False):
+        if new_box:
+            self.box = QtWidgets.QGraphicsRectItem(self.start_x, self.start_y, 0, 0)
```
Comment (Tamookk) [inline src/View/mainpage/DrawROIWindow/DrawBoundingBox.py:24]:
```suggestion
            self.box = QtWidgets.QGraphicsRectItem(self.start_x, self.start_y,
                                                   0, 0)
```

PEP 8 compliance. Also, all functions in this file need docstrings.

File: src/View/mainpage/DrawROIWindow/Drawing.py [hunk 6]
```diff
@@ -0,0 +1,291 @@
+import math
```
Comment (Tamookk) [inline src/View/mainpage/DrawROIWindow/Drawing.py:1]: Many lines in this file far too long for PEP 8 compliance:
14-18, 22-23, 26, 47, 60, 66-69, 77, 85-86, 90-91, 107, 125, 133, 136, 141, 143, 146, 160, 163, 170, 172, 178, 185-186, 191-192, 196, 214, 219, 225, 233, 238, 266-267, 270, 272, 280, 282, & 284.

Also, all functions either don't have docstrings or have incorrect docstrings.

File: src/View/mainpage/DrawROIWindow/SaveROIProgressWindow.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
+import pydicom
+from PySide6 import QtWidgets, QtCore
+
+from src.Model import ROI
+from src.Model.Worker import Worker
+
+
+class SaveROIProgressWindow(QtWidgets.QDialog):
+    """
+    This class displays a window that advises the user that the RTSTRUCT is being modified, and then creates a new
```
Comment (Tamookk) [inline src/View/mainpage/DrawROIWindow/SaveROIProgressWindow.py:10]: PEP 8 compliance, line too long. Also line 41.

File: src/constants.py [hunk 8]
```diff
@@ -1,3 +1,4 @@
 DEFAULT_WINDOW_SIZE = 512
 INITIAL_ONE_VIEW_ZOOM = 1
 INITIAL_FOUR_VIEW_ZOOM = 0.5
+INITIAL_DRAWING_TOOL_RADIUS = 19
```
Comment (Tamookk) [inline src/constants.py:4]: No newline at end of file - PEP 8 compliance.

File: src/View/mainpage/DrawROIWindow/SelectROIPopUp.py [hunk 9]
```diff
@@ -0,0 +1,115 @@
+import csv
+import platform
+
+from PySide6 import QtCore, QtGui
+from PySide6.QtWidgets import QDialog, QLabel, QLineEdit, QWidget, QPushButton, QHBoxLayout, QListWidget, QVBoxLayout
```
Comment (Tamookk) [inline src/View/mainpage/DrawROIWindow/SelectROIPopUp.py:5]: Line too long for PEP 8 compliance. Also lines:
10-14, 34, & 73

Also, class and functions have no docstrings.

File: src/View/mainpage/DrawROIWindow/UIDrawROIWindow.py [hunk 10]
```diff
@@ -1,48 +1,45 @@
...
-import numpy
 import pydicom
 from PySide6 import QtCore, QtGui, QtWidgets
-from PySide6.QtCore import QSize, Qt
-from PySide6.QtGui import QIcon, QPixmap, QColor, QPen
-from PySide6.QtWidgets import QMessageBox, QHBoxLayout, QLineEdit, QSizePolicy, QPushButton, QDialog, QListWidget, \
-    QGraphicsPixmapItem, QGraphicsEllipseItem, QVBoxLayout, QLabel, QWidget, QFormLayout
+from PySide6.QtCore import Qt, QSize
+from PySide6.QtGui import QIcon, QPixmap
+from PySide6.QtWidgets import QFormLayout, QLabel, QLineEdit, QSizePolicy, QHBoxLayout, QPushButton, QWidget, \
```
Comment (Tamookk) [inline src/View/mainpage/DrawROIWindow/UIDrawROIWindow.py:7]: Line too long for PEP 8 compliance. Also lines:
18, 36, 127, 144, 147, 151, 153-155, 158, 162, 164, 166, 168, 215, 224-231, 295-298, 437, 453, 464, 470, 485, 495, 497, 537, 540, 617, 645, 659-664, 678, 720-721, 752, 758, 763-768, and most of the most of the `retranslate_ui`, `init_cursor_radius_change_box`, `disable_cursor_radius_change_box`, `enable_cursor_radius_change_box`, and `display_multipolygon_warning` functions.

Also too big of a gap between lines 277 and 282, and lines 555 and 559 - should only be one blank line according to PEP 8. Many functions do not have docstrings or have incorrect/incomplete docstrings too.

File: test/test_model_roi.py [hunk 11]
```diff
@@ -102,7 +102,7 @@ def test_add_to_roi():
     patient_dict_container.set_initial_values(None, None, None, blah="blah", rois={})
     if patient_dict_container.get("rois") is not None:
         print("rois are present in patient dict container")
-    updated_rtss = create_roi(rt_ss, roi_name, roi_coordinates, image_ds)
+    updated_rtss = create_roi(rt_ss, roi_name, [{'coords': roi_coordinates, 'ds': image_ds}])
```
Comment (Tamookk) [inline test/test_model_roi.py:105]: Line too long for PEP 8 compliance, also line 143.

File: test/test_view_structures_tab.py [hunk 12]
```diff
@@ -185,11 +185,11 @@ def test_merge_rtss(qtbot, test_object):
     # the old rtss
     roi_name = test_object.rois.get(1)["name"]
     roi_coordinates = [0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0]
-    new_rtss = create_roi(new_rtss, roi_name, roi_coordinates, dataset)
+    new_rtss = create_roi(new_rtss, roi_name, [{'coords': roi_coordinates, 'ds': dataset}])
```
Comment (Tamookk) [inline test/test_view_structures_tab.py:188]: Line too long for PEP 8 compliance, also line 192.

**Review by Tamookk — COMMENTED** [review]
Encountered no issues when testing the code. Most of my feedback is to do with PEP 8 compliance, mostly with line lengths. If using PyCharm, it has the ability to add visual guides in Settings > Editor > Code Style, which can help with keeping lines within the 79 character limit. PyCharm will also suggest other changes relating to the PEP 8 standard, such as blank lines, docstrings, etc. 

Overall good work.

**Review by JThomas2020 — APPROVED** [review]
Team 19, encountered no errors during testing, @Tamookk has covered all PEP-8 and docstrings, cheers for that! Good Job guys!

**Review by didymo — APPROVED** [review]

File: src/Model/ISO2ROI.py (hunk 1 above)
Comment (eddieGong2604) [reply src/Model/ISO2ROI.py:195]: Resolved. Can you please check?

File: src/Model/ROI.py (hunk 2 above)
Comment (eddieGong2604) [reply src/Model/ROI.py:137]: Resolved. Can you please check?

File: src/View/mainpage/DrawROIWindow/UIDrawROIWindow.py (hunk 10 above)
Comment (eddieGong2604) [reply src/View/mainpage/DrawROIWindow/UIDrawROIWindow.py:7]: Resolved. Can you please check?

File: test/test_view_structures_tab.py (hunk 12 above)
Comment (eddieGong2604) [reply test/test_view_structures_tab.py:188]: Resolved. Can you please check?

**Review by Tamookk — APPROVED** [review]
Team 14

**Pull request merged**